<?xml version="1.0"?>
<launch>
    <!-- <node pkg="permobil_arm" type="pcl_converter" name="pcl_converter" output="screen" />

    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_to_pointcloud" output="screen" launch-prefix="bash -c 'sleep 1; $0 $@' ">
        <param name="interval" value="0.1" />
        <param name="file_name" value="$(find moveit_task_constructor_gpd)/data/pointclouds/debug.pcd" />
        <param name="frame_id" value="/link_base" />
        <remap from="/cloud_pcd" to="/gpd_fullpcd" />
    </node> -->
    <!-- get rid of lone points-->
    <node pkg="nodelet" type="nodelet" name="outlier" args="load pcl/StatisticalOutlierRemoval pcl_manager" output="screen">
        <remap from="~input" to="camera/depth/color/points" />
        <remap from="~output" to="gpd_converter1" />
        <param name="~mean_k" value="10" />
        <param name="~stddev" value="1" />
        <param name="~negative" value="False" />
    </node>
    <!-- Limit range -->
    <node pkg="nodelet" type="nodelet" name="passthrough1" args="load pcl/PassThrough pcl_manager" output="screen">
        <remap from="~input" to="gpd_converter1" />
        <remap from="~output" to="gpd_converter2" />
        <rosparam>
filter_field_name: z
filter_limit_min: 0
filter_limit_max: 1
filter_limit_negative: False
input_frame: link_tcp
</rosparam>
    </node>


    <node pkg="nodelet" type="nodelet" name="passthrough2" args="load pcl/PassThrough pcl_manager" output="screen">
        <remap from="~input" to="gpd_converter2" />
        <remap from="~output" to="gpd_converter3" />
        <rosparam>
filter_field_name: y
filter_limit_min: -1
filter_limit_max: 1
filter_limit_negative: False
</rosparam>
    </node>

    <node pkg="nodelet" type="nodelet" name="passthrough3" args="load pcl/PassThrough pcl_manager" output="screen">
        <remap from="~input" to="gpd_converter3" />
        <remap from="~output" to="gpd_pcd" />
        <rosparam>
  filter_field_name: x
  filter_limit_min: -1
  filter_limit_max: 1
  filter_limit_negative: False
  output_frame: link_base
</rosparam>
    </node>


</launch>